test/controllers/renewal/HowCashPaymentsReceivedControllerSpec.scala

```diff
@@ -0,0 +1,110 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.renewal
+
+import connectors.DataCacheConnector
+import models.renewal.{CashPayments, CashPaymentsCustomerNotMet, HowCashPaymentsReceived, PaymentMethods, Renewal}
+import org.jsoup.Jsoup
+import org.mockito.Matchers._
+import org.mockito.Mockito.when
+import play.api.test.Helpers._
+import play.api.test.Helpers.{contentAsString, status}
+import services.RenewalService
+import uk.gov.hmrc.http.cache.client.CacheMap
+import utils.{AmlsSpec, AuthorisedFixture}
+
+import scala.concurrent.Future
+
+class HowCashPaymentsReceivedControllerSpec extends AmlsSpec {
+
+  lazy val mockDataCacheConnector = mock[DataCacheConnector]
+  lazy val mockRenewalService = mock[RenewalService]
+
+  val receiveCashPayments = CashPayments(CashPaymentsCustomerNotMet(true), Some(HowCashPaymentsReceived(PaymentMethods(true,true,Some("other")))))
+  val doNotreceiveCashPayments = CashPayments(CashPaymentsCustomerNotMet(false), None)
+
+  trait Fixture extends AuthorisedFixture {
+    self => val request = addToken(authRequest)
+
+    val controller = new HowCashPaymentsReceivedController (
+      dataCacheConnector = mockDataCacheConnector,
+      authConnector = self.authConnector,
+      renewalService = mockRenewalService
+    )
+
+    when(mockRenewalService.getRenewal(any(),any(),any()))
+      .thenReturn(Future.successful(None))
+
+    when(mockRenewalService.updateRenewal(any())(any(),any(),any()))
+      .thenReturn(Future.successful(new CacheMap("", Map.empty)))
+  }
+
+  "HowCashPaymentsReceived" when {
+    "get is called" must {
+      "load the page" when {
+        "renewal data is found for how cash payments received and pre-populate the data" in new Fixture {
+          when(mockRenewalService.getRenewal(any(),any(),any()))
+            .thenReturn(Future.successful(Some(Renewal(receiveCashPayments = Some(receiveCashPayments)))))
+
+          val result = controller.get()(request)
+          status(result) mustEqual OK
+
+          val page = Jsoup.parse(contentAsString(result))
+
+          page.select("input[type=checkbox][name=cashPaymentMethods.courier][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=checkbox][name=cashPaymentMethods.direct][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=checkbox][name=cashPaymentMethods.other][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=text][name=cashPaymentMethods.details]").first().`val`() must be("other")
+        }
+
+        "no renewal data is found and show an empty form" in new Fixture {
```

Comment:
`no renewal data is found it shows an empty form` ?

Reply:
i'll change that